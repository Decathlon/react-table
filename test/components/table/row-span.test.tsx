/// <reference path="../../typings/tests-entry.d.ts" />
import * as React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import { cleanup, fireEvent } from "@testing-library/react";

import RowSpan from "../../../src/components/table/row-span";
import { customRender } from "../../tests-utils/react-testing-library-utils";

describe("RowSpan component", () => {
  test("should render a closed span", () => {
    const props = {
      opened: false,
      length: 2,
      title: "Foo",
      color: "red",
      toggle: () => null
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<RowSpan {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render an opened span", () => {
    const props = {
      opened: true,
      length: 2,
      title: "Foo",
      toggle: () => null
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<RowSpan {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should call the span toggle callback", () => {
    const props = {
      opened: true,
      length: 2,
      title: "Foo",
      toggle: jest.fn()
    };
    const { container } = customRender(<RowSpan {...props} />);
    fireEvent.click(container.getElementsByTagName("button")[0]);
    expect(props.toggle).toBeCalledTimes(1);
    cleanup();
  });
});
