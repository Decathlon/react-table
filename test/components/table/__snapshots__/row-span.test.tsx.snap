// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowSpan component should render a closed span 1`] = `
<td
  className="table-column row-span-column"
  rowSpan={2}
  style={
    Object {
      "borderLeft": "solid 15px red",
      "maxWidth": 110,
      "minWidth": 110,
      "width": 110,
    }
  }
>
  <div
    className="row-span-container"
  >
    <div
      className="row-span-text"
    >
      Foo
    </div>
    <ForwardRef(WithStyles)
      data-testid="table-toggle-row-btn"
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)>
        keyboard_arrow_right
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>
</td>
`;

exports[`RowSpan component should render an opened span 1`] = `
<td
  className="table-column row-span-column"
  rowSpan={2}
  style={
    Object {
      "borderLeft": "solid 15px initial",
      "maxWidth": 110,
      "minWidth": 110,
      "width": 110,
    }
  }
>
  <div
    className="row-span-container"
  >
    <div
      className="row-span-text"
    >
      Foo
    </div>
    <ForwardRef(WithStyles)
      data-testid="table-toggle-row-btn"
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)>
        keyboard_arrow_down
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>
</td>
`;
